### ISSUE TITLE
VGA output?

### ISSUE THREAD

**Issue by dylanmc** [MiSTer-devel/mr-fusion#6]
I see that mr-fusion is focused on HDMI, but I was wondering if it's possible to enable VGA output. I bought an Ultimate MiSTer, and its card came configured to drive both HDMI and VGA, but I can't figure out how to back-up or create a new microSD card that does the same. Naturally, I thought mr-fusion was the right way to start, but ... Hmm.

**Comment by michaelshmitty** [comment]
Hi @dylanmc ,

That's right, mr-fusion is just a very tiny linux distribution that does nothing more than boot the DE10-Nano board, wipe the SD card and installs the standard MiSTer release on it. When it restarts mr-fusion is completely gone.

To back up your original SD card, use a utility like [Win32 Disk Imager](https://sourceforge.net/projects/win32diskimager/) or [ApplePi-Baker](https://www.tweaking4all.com/software/macosx-software/applepi-baker-v2/). It's a similar process as with a Raspberry Pi SD card (https://pimylifeup.com/backup-raspberry-pi/).

The DE10-Nano has no onboard display hardware, i.e. a graphics card. It can only output to a serial console. Mr-fusion contains a sample HDMI hardware implementation (core) that gets loaded onto the FPGA at boot time. The Linux kernel is configured to use it for display output. I reused this implementation from example code, I did not implement this my self as my FPGA skills are severely lacking.
It would take someone with good FPGA skills to amend the simple HDMI core that mr-fusion uses to output a VGA signal over the HDMI connector that could then be used with a HDMI to VGA dongle. Or to implement the part in MiSTer that outputs VGA over the I/O board.
It's not  in the scope of mr-fusion in the foreseeable future.
